pkg/proxy/iptables/proxier.go

```diff
@@ -248,6 +251,34 @@ func (l *listenPortOpener) OpenLocalPort(lp *utilproxy.LocalPort) (utilproxy.Clo
 // Proxier implements proxy.Provider
 var _ proxy.Provider = &Proxier{}
 
+type detectLocalByCIDR struct {
+	cidr string
+}
+
+// NewDetectLocalByCIDR implements the LocalTrafficDetector interface using cluster CIDR.
+func NewDetectLocalByCIDR(cidr string, ipt utiliptables.Interface) (utiliptables.LocalTrafficDetector, error) {
```

Comment:
I can't immediately see how this impl is different from the ipvs impl.  If they are the same, why not put one impl into `pkg/proxy/util/iptables` and both use it?

Reply:
I originally thought they might be different, but having refactored the args out, you are right, they became essentially the same. Can refactor to one as you pointed out and move to a util package.

Reply:
Actually now I understand why. It uses the package local writeLine function. Will have to export that too. I guess I was following the existing pattern of duplication. Will refactor.

Reply:
Why not make this return `args []string` and let the caller join them?

Reply:
good point :-)